notesystem/common/config.py

```diff
@@ -0,0 +1,512 @@
+
+        Returns:
+            {argparse.ArgumentParser} - The (created) argument parser
+
+        """
+
+        OPTIONS = self.OPTIONS
+        parser = argparse.ArgumentParser(
+            prog='notesystem',
+            description='A system for handing your notes for you',
+        )
+        # Create the convert and check parsers
+        mode_parser = parser.add_subparsers(
+            title='modes',
+            description='choose a mode to use (required)',
+            help='use {convert, check} --help to view more options',
+        )
+        convert_parser = mode_parser.add_parser(
+            'convert',
+            help='convert files from markdown to html (by default) or to pdf \
+                 (using --to-pdf)',
+        )
+        convert_parser.set_defaults(mode='convert')
+        check_parser = mode_parser.add_parser(
+            'check',
+            help='check markdown file(s) for styling errors',
+        )
+        check_parser.set_defaults(mode='check')
+
+        # Parse the OPTIONS dict and create the argparser
+        for section in OPTIONS:
+            if section == 'general':
+                # Create the main parser
+                for option in OPTIONS[section]:
+                    opts = self._gen_argparse_args(
+                        OPTIONS[section][option],
+                    )
+                    parser.add_argument(*opts[0], **opts[1])
+            elif section == 'convert':
+                for option in OPTIONS[section]:
+                    opts = self._gen_argparse_args(
+                        OPTIONS[section][option],
+                    )
+                    convert_parser.add_argument(*opts[0], **opts[1])
+            elif section == 'check':
+                for option in OPTIONS[section]:
+                    # TODO: Should this only be in check?
+                    #       Currenlty only the disabled_errors needs this
+                    #       but in the future more options might?
+                    # Handle disabled errors (is a list)
+                    if isinstance(OPTIONS[section][option], list):
+                        op = OPTIONS[section][option]
+                        if len(op) < 1:
+                            continue
+                        assert 'group_name' in op[0]
+                        assert 'group_desc' in op[0]
+                        new_group = check_parser.add_argument_group(
+                            op[0]['group_name'],
+                            op[0]['group_desc'],
+                        )
+                        for i in OPTIONS[section][option]:
+                            opts = self._gen_argparse_args(i)
+                            new_group.add_argument(*opts[0], **opts[1])
+                    else:
+                        opts = self._gen_argparse_args(
+                            OPTIONS[section][option],
+                        )
+                        check_parser.add_argument(*opts[0], **opts[1])
```

Comment:
```suggestion
                        args, kwargs = self._gen_argparse_args(
                            OPTIONS[section][option],
                        )
                        check_parser.add_argument(*args, **kwargs)
```

This makes it more clear what is happening.

Reply:
Will do